### PR TITLE
W-16814280: fix plugin class loader leak

### PR DESCRIPTION
**Pull request by rbourbonnavarro**

### CODE REVIEW

**Comment by rbourbonnavarro** [comment]
--validate

**Comment by rbourbonnavarro** [comment]
--validate

File: modules/deployment/src/test/java/org/mule/runtime/module/deployment/test/internal/XmlPluginClassLoaderLeakOnDeploymentTestCase.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    XmlExtensionLoaderDelegate.forceTransformerPoolRecreation(true);
+  }
+
+  @AfterClass
+  public static void after() {
+    XmlExtensionLoaderDelegate.forceTransformerPoolRecreation(false);
+  }
+
+  public static final Supplier<Set<ArtifactPluginFileBuilder>> XML_SDK_PLUGIN =
+      () -> new HashSet<>(singletonList(byeXmlExtensionPlugin));
```
Comment (elrodro83) [inline modules/deployment/src/test/java/org/mule/runtime/module/deployment/test/internal/XmlPluginClassLoaderLeakOnDeploymentTestCase.java:47]: just `singleton()`?

File: modules/extensions-xml-support/src/main/java/org/mule/runtime/extension/internal/loader/xml/XmlExtensionLoaderDelegate.java [hunk 2]
```diff
@@ -250,13 +252,41 @@ private static ParameterRole getRole(final String role) {
...
+   * {@code FOR_TNS_XSTL_TRANSFORMER_POOL}.
+   */
+  private static boolean forceTransformerPoolRecreation;
+
+  /**
+   * Only for testing purposes.
+   *
+   * @param force whether to force the recreation of the transformer pool {@code FOR_TNS_XSTL_TRANSFORMER_POOL}.
+   */
+  public static void forceTransformerPoolRecreation(boolean force) {
```
Comment (elrodro83) [inline modules/extensions-xml-support/src/main/java/org/mule/runtime/extension/internal/loader/xml/XmlExtensionLoaderDelegate.java:268]: add as depecated

**Review by elrodro83 — CHANGES_REQUESTED** [review]

**Review by elrodro83 — APPROVED** [review]

**Pull request merged**